acceptance: TestDockerCLI_test_socket_name failed

acceptance.TestDockerCLI_test_socket_name [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/16040322?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_Acceptance/16040322?buildTab=artifacts#/) on master @ [2ee6332c73b7469719610b36bf73b1101fdc14f0](https://github.com/cockroachdb/cockroach/commits/2ee6332c73b7469719610b36bf73b1101fdc14f0):


```
=== RUN   TestDockerCLI_test_socket_name
    test_log_scope.go:170: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/c08ff262c750a522d8c02a8762e12dc5/logTestDockerCLI_test_socket_name3840231649
    test_log_scope.go:81: use -show-logs to present logs inline
    generated_cli_test.go:411: -- test log scope end --
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/_tmp/c08ff262c750a522d8c02a8762e12dc5/logTestDockerCLI_test_socket_name3840231649
--- FAIL: TestDockerCLI_test_socket_name (61.59s)
=== RUN   TestDockerCLI_test_socket_name/test_socket_name.tcl
    util_docker.go:260: non-zero exit code: 1
    --- FAIL: TestDockerCLI_test_socket_name/test_socket_name.tcl (54.34s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDockerCLI_test_socket_name.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/cli/interactive_tests/test_socket_name.tcl
+++ b/pkg/cli/interactive_tests/test_socket_name.tcl
@@ -13,7 +13,7 @@ send "mkdir -p $longname\r"
 eexpect ":/# "
 
 start_test "Check derived Unix socket name is printed in server output"
-send "$argv start-single-node --insecure --listen-addr=`cat /etc/hostname`:0 --pid-file=logs/server_pid --socket-dir=.\r"
+send "$argv start-single-node --insecure --listen-addr=127.0.0.1:0 --pid-file=logs/server_pid --socket-dir=.\r"
 eexpect "CockroachDB node starting"
 expect {
     -re "sql:.*@\[^:\]*:(\[^/\]*)/" { set sql_port $expect_out(1,string) }
@@ -33,17 +33,27 @@ send "PS1=':''/# '\r"
 eexpect ":/# "
 
 start_test "Check that the socket is locked from reuse while the server is running."
-# We use 127.0.0.1 so as to not overlap with the default address 0.0.0.0 selected above.
+# Use 127.0.0.2 (different from 127.0.0.1 above) so the TCP bind succeeds and
+# the test isolates the Unix socket lock behavior. Both servers share the same
+# --socket-dir and port, so the socket filename is the same.
 # We also need a different data directory to avoid a conflict there.
-send "$argv start-single-node --insecure --listen-addr=127.0.0.1:$sql_port --http-addr=:0 --socket-dir=. -s=path=logs/other\r"
+send "$argv start-single-node --insecure --listen-addr=127.0.0.2:$sql_port --http-addr=:0 --socket-dir=. -s=path=logs/other\r"
 eexpect ERROR
 eexpect "Socket appears locked by process"
 eexpect ":/# "
 end_test
 
 start_test "Check that stopping the first process abruptly enables the 2nd process to start"
-system "kill -9 `cat logs/server_pid`"
-send "$argv start-single-node --insecure --listen-addr=127.0.0.1:$sql_port --http-addr=:0 --socket-dir=. -s=path=logs/other\r"
+# Kill the server and wait for the process to fully exit so the Unix socket
+# lock file becomes reclaimable by the replacement server.
+system "PID=\$(cat logs/server_pid); kill -9 \$PID
+        for i in \$(seq 1 30); do
+          kill -0 \$PID 2>/dev/null || exit 0
+          sleep 0.1
+        done
+        echo 'process still alive'
+        exit 1"
+send "$argv start-single-node --insecure --listen-addr=127.0.0.2:$sql_port --http-addr=:0 --socket-dir=. -s=path=logs/other\r"
 eexpect "CockroachDB node starting"
 system "test -S .s.PGSQL.$sql_port"
 system "test -r .s.PGSQL.$sql_port.lock"

